### PR TITLE
[15.05] UI, Collections: use Galaxy.options.root for API URL

### PR DESCRIPTION
**Pull request by carlfeberhard**
(see mailing list: [galaxy-dev] Possible bug on "Create a collection of paired datasets":
http://dev.list.galaxyproject.org/Possible-bug-on-quot-Create-a-collection-of-paired-datasets-quot-td4667510.html)

### CODE REVIEW

**Comment by jmchilton** [comment]
:+1: - great! Thanks for this - I tried to hunt this bug down myself yesterday for a little while and failed pretty badly.

**Comment by blankenberg** [comment]
+1

File: client/galaxy/scripts/mvc/collection/list-of-pairs-collection-creator.js [hunk 1]
```diff
@@ -568,7 +568,8 @@ var PairedCollectionCreator = Backbone.View.extend( baseMVC.LoggableMixin ).exte
      */
     createList : function( name ){
         var creator = this,
-            url = '/api/histories/' + this.historyId + '/contents/dataset_collections';
+            root = ( window.Galaxy && Galaxy.options.root )? Galaxy.options.root : '/',
```
Comment (bgruening) [inline client/galaxy/scripts/mvc/collection/list-of-pairs-collection-creator.js:571]: Just out of interest, shouldn't this be offered as a function `get_galaxy_root()`?

Comment (carlfeberhard) [inline client/galaxy/scripts/mvc/collection/list-of-pairs-collection-creator.js:571]: Where would that function live as far as scope? If you make it window, you'd still need the ternary operator (outside the function): 
`root = window.get_galaxy_root? get_galaxy_root() : '/',`
Is that what your were hoping to clear up or something else?

The `Galaxy` object should be available on every page at this point, but it doesn't hurt to do the fall back as a bit of defensive programming.

Comment (bgruening) [inline client/galaxy/scripts/mvc/collection/list-of-pairs-collection-creator.js:571]: > Where would that function live as far as scope? If you make it window, you'd still need the ternary operator (outside the function):
> root = window.get_galaxy_root? get_galaxy_root() : '/',
> Is that what your were hoping to clear up or something else?

Yes, this one.

> The Galaxy object should be available on every page at this point, but it doesn't hurt to do the fall back as a bit of defensive programming.

Ok, makes sense. Thanks for the explanation.

**Pull request merged**